shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -342,6 +399,124 @@ void setTextInputEditingState(View view, TextInputChannel.TextEditState state) {
     }
   }
 
+  private interface MinMax {
+    void inspect(double x, double y);
+  }
+
+  private void saveEditableSizeAndTransform(double width, double height, double[] matrix) {
+    final double[] minMax = new double[4]; // minX, maxX, minY, maxY.
+    final boolean isAffine = matrix[3] == 0 && matrix[7] == 0 && matrix[15] == 1;
+    minMax[0] = minMax[1] = matrix[12] / matrix[15]; // minX and maxX.
+    minMax[2] = minMax[3] = matrix[13] / matrix[15]; // minY and maxY.
+
+    final MinMax finder =
+        new MinMax() {
+          @Override
+          public void inspect(double x, double y) {
+            final double w = isAffine ? 1 : 1 / (matrix[3] * x + matrix[7] * y + matrix[15]);
+            final double tx = (matrix[0] * x + matrix[4] * y + matrix[12]) * w;
+            final double ty = (matrix[1] * x + matrix[5] * y + matrix[13]) * w;
+
+            if (tx < minMax[0]) minMax[0] = tx;
+            else if (tx > minMax[1]) minMax[1] = tx;
+
+            if (ty < minMax[2]) minMax[2] = ty;
+            else if (ty > minMax[3]) minMax[3] = ty;
```

Comment:
I realized the code is missing braces in a few places. The Google style guide for Java suggests that braces are used even if there's only a single statement: https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

Reply:
Thank you for reviewing this!

Reply:
All missing braces should be added now. Does the updated code look good to you?